nighthawk/source/common/rate_limiter_impl.cc

```diff
@@ -0,0 +1,32 @@
+#include "nighthawk/source/common/rate_limiter_impl.h"
+
+#include "common/common/assert.h"
+
+#include "nighthawk/common/exception.h"
+
+namespace Nighthawk {
+
+LinearRateLimiter::LinearRateLimiter(Envoy::TimeSource& time_source, const Frequency frequency)
+    : time_source_(time_source), acquireable_count_(0), acquired_count_(0), frequency_(frequency),
+      started_at_(time_source_.monotonicTime()) {
+  ASSERT(frequency.value() > 0, "Frequency must be > 0");
+}
+
+bool LinearRateLimiter::tryAcquireOne() {
+  if (acquireable_count_ > 0) {
```

Comment:
I do worry about seeing bursty behavior still, even though we are now continuous and linear. This is more a non-actionable "something to think about", but I know from work on `httprr`, the interaction of thread scheduling and rate limiters like this can be quite jerky.